Packs/Core/Integrations/CortexCoreIR/CortexCoreIR.py

```
@@ -87,6 +98,106 @@ def create_indicator_rule_request(self, request_data: Union[dict, str], suffix:
+        If endpoint disconnected/not exists the group id will be None.
+        Args:
+            endpoint_id (str): ID of the endpoint to apply the block.
+            ip_list (list[str]): IP addresses to block.
+            duration (int): Block duration in seconds.
+
+        Returns:
+            list[dict]: A list of action records, each containing:
+                - ip_address (str): The blocked IP.
+                - endpoint_id (str): The endpoint where the block was applied.
+                - group_id (str): ID of the block action for status polling.
+        """
+        results = []
+        if not self._is_endpoint_connected(endpoint_id):
+            demisto.debug(f"Cannot block ip list. Endpoint {endpoint_id} is not connected.")
+            return [{"ip_address": ip_address, "group_id": None, "endpoint_id": endpoint_id} for ip_address in ip_list]
+        for ip_address in ip_list:
+            result = {
+                "ip_address": ip_address,
+                "endpoint_id": endpoint_id,
+            }
+            if is_ipv6_valid(ip_address) or is_ip_valid(ip_address):
+                demisto.debug(f"Blocking ip address: {ip_address}")
+                response = self._http_request(
+                    method="POST",
+                    headers=self._headers,
+                    url_suffix="/endpoints/block_ip",
+                    json_data={
+                        "request_data": {
+                            "addresses": [ip_address],
+                            "endpoint_id": endpoint_id,
+                            "direction": "both",
+                            "duration": duration,
+                        }
+                    },
+                )
+                group_id = response.get("reply", {}).get("group_action_id")
+                demisto.debug(f"Block request for {ip_address} returned with group_id {group_id}")
+                result["group_id"] = group_id
+            else:
```

We should throw an error for this case